### ISSUE TITLE
image push: Can't push to dockerhub by using default docker credentials

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [inspektor-gadget/inspektor-gadget#2299]
Log to docker.io by using docker login:

```bash 
$ docker login ....
...
Login Succeeded

$ cat /home/mvb/.docker/config.json
{
	"auths": {
		"https://index.docker.io/v1/": {
			"auth": "foo"
		}
	}
}
```

Credentials file for IG doesn't exist:

```bash
$ sudo cat /var/lib/ig/config.json
cat: /var/lib/ig/config.json: No such file or directory
```

I can't push an image to dockerhub:

```bash 
$ sudo -E ig image push docker.io/mauriciovasquezbernal/mygadget:latest -v
INFO[0000] Experimental features enabled                
Pushing docker.io/mauriciovasquezbernal/mygadget:latest...
DEBU[0000] Using auth file "/var/lib/ig/config.json"    
DEBU[0000] Couldn't find default auth file "/var/lib/ig/config.json"... 
DEBU[0000] Using default docker auth file instead       
DEBU[0000] $HOME: "/home/mvb"                           
Error: pushing gadget: copying to remote repository: PUT "https://registry-1.docker.io/v2/mauriciovasquezbernal/mygadget/manifests/latest": response status code 401: unauthorized: authentication required: [map[Action:pull Class: Name:mauriciovasquezbernal/mygadget Type:repository] map[Action:push Class: Name:mauriciovasquezbernal/mygadget Type:repository]]
```

If I use `ig login`, it works fine:

```bash
$ sudo -E ig login docker.io
INFO[0000] Experimental features enabled                
Username: mauriciovasquezbernal
Password: 
Login Succeeded!

$ sudo cat /var/lib/ig/config.json
{
	"auths": {
		"docker.io": {
			"auth": "foo"
		}
	}
}

$ sudo -E ig image push docker.io/mauriciovasquezbernal/mygadget:latest -v
INFO[0000] Experimental features enabled                
Pushing docker.io/mauriciovasquezbernal/mygadget:latest...
DEBU[0000] Using auth file "/var/lib/ig/config.json"    
Successfully pushed docker.io/mauriciovasquezbernal/mygadget:latest@sha256:63d13942ad850d126d6e4abce3788f1e2d4709d12979d77d34026f4555487a76
```

I think the main issue is that the host is saved differently in both files:
- /home/mvb/.docker/config.json -> https://index.docker.io/v1/
- /var/lib/ig/config.json -> docker.io

Then https://github.com/inspektor-gadget/inspektor-gadget/blob/1573fcef57920e2d247159806467c091083ed57c/pkg/oci/oci.go#L380
returns an empty config

**Comment by mauriciovasquezbernal** [comment]
This bug is still present as of today.

**Comment by ask-elad** [comment]
hey @mauriciovasquezbernal 
i tried to replicate this issue locally what i found was this 

<img width="804" height="203" alt="Image" src="https://github.com/user-attachments/assets/107c9913-80d9-44d4-ae22-dadfda5a74da" />

docker-credential-desktop this is being used by docker to store the credentials secure at some other place
Now the probelm you mentioned is also present as the GetAuthConfig isn't checking for aliases correctly

What i was thinking was instead of using GetAuthConfig we can implement our own function which basically check for aliases as well
so for example if the hoststring is docker instead of checking for just docker that function checks for other possiblities like index.docker.io registry-1.docker.io as well

I wanted to know you opinion on this before working on this issue

**Comment by mauriciovasquezbernal** [comment]
I think that makes sense!

**Comment by ask-elad** [comment]
alright then assign this issue to me i would like to work on this issue!

**Comment by ask-elad** [comment]
i worked on this a little and i need some guidance as how to proceed 

1)  initially i thought of implementing an own function that checks for aliases as well but i figured out that since the possiblities of corresponding hoststring is low instead of creating a new function from scratch we can actually use the existing docker function by iterating all possible hoststrings and passing each one of them till we get a match

so i moved ahead with this approach 

<img width="1427" height="391" alt="Image" src="https://github.com/user-attachments/assets/c07b36c0-c08a-4e1a-a572-c234bcdb6d58" />

but then i hit a wall as Docket doesn't stores it's credentials directly in ~/.dockerconfig.json
this can be fixed as when logging into docker we would have to login like this 
docker login --username username --password <token>

this basically forces docker to store plaintext creds into auth instead of helpers 
but i figured that would be inconvenient to users

so in short our ig function shall be able to access the helper functions that docker is using to store the creds, but i am not sure about this approach and i am little confused about it's implementation 

i would like to know your thoughts
